libs/tree-sitter-titan/grammar.js

```diff
@@ -1,7 +1,42 @@
 module.exports = grammar({
-  name: 'titan',
-
+  name: "titan",
   rules: {
-    source_file: $ => 'hello'
-  }
-});
\ No newline at end of file
+    variable_definition: ($) =>
+      seq(
+        $._variable_start,
+        field("pattern", $.alpha_identifier),
+        optional(seq(":", field("type", $._type))),
+        "=",
+        field("value", $._non_null_literal)
+      ),
+
+    // TODO: Declare without assign
+    variable_declaration: ($) => seq(),
+
+    alpha_identifier: () =>
+      /[A-Za-z\$_][A-Z\$_a-z0-9]*(_[\-!#%&*+\/\\:<=>?@\u005e\u007c~]+)?/,
+
+    // TODO: Add annotations and modifiers and stuff
+    _variable_start: ($) => seq("let"),
+
+    // TODO: Add other types, maybe function/lambda
+    _type: ($) => choice($.literal_type),
```

Comment:
Maybe we make this just a list of supported types until we add support for UserDefined types.

so in the meantime we dont allow 
```js
let x: Stringzs = 10
```
even though `Stringzs` could be a user defined class or interface etc... Atleast until we map out how we are going to handle user types.

So for no maybe just allow `string`, `num`, `bool`.

Reply:
```suggestion
    _type: ($) => choice("string", "int", "bool"), // choice($.literal_type),
```